lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -523,18 +523,7 @@ function dosomething_reportback_view_entity($entity) {
     $fid = array_pop($entity->fids);
   }
 
-
-  // @TODO Can we templatize/DRY some of this? It's fairly similiar code in multiple places.
-  $disable_reactions = dosomething_kudos_disable_reactions($entity->nid);
-
-  if (! $disable_reactions) {
-    $kudos = [];
-    $kudos['rbid'] = $entity->rbid;
-    $kudos['disable_reactions'] = $disable_reactions;
-    $kudos['allowed_reactions'] = [dosomething_kudos_get_term_id_by_name('heart')];
-    $kudos['reaction_totals'] = ['heart' => dosomething_kudos_get_total_for_file_by_term_name($entity->rbid, 'heart')];
-    $kudos['existing_kids'] = dosomething_kudos_get_kudos_for_user_by_file($entity->rbid);
-  }
```

Comment:
Was it a problem before that the `$kudos` var would only be created if reactions weren't disabled... but then at the bottom the `$kudos` var was passed regardless of whether it was created or not? I wonder if the check for disabled reactions can just be on its own like:

```php
$kudos = NULL;  // or []

if (! dosomething_kudos_disable_reactions($entity->nid)) {
  $kudos = dosomething_kudos_get_kudos_data_for_fid($fid);
}
```